From 0594ba53b9b8d9a1ac409fd187e4d1ba8f2e7f2a Mon Sep 17 00:00:00 2001
From: Vineet Gupta <vgupta@synopsys.com>
Date: Thu, 8 Aug 2013 15:41:51 +0530
Subject: [PATCH] ARC: SuSv3 legacy support: Add "bcmp" alias for memcmp

Although uClibc provides this already as UCLIBC_SUSV3_LEGACY_MACROS,
that however requires UCLIBC_SUSV3_LEGACY to be switched off, causing a
bunch of other things to be not available (usleep)

Signed-off-by: Vineet Gupta <vgupta@synopsys.com>
---
 libc/string/arc/memcmp.S | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libc/string/arc/memcmp.S b/libc/string/arc/memcmp.S
index 0c7c345..cb25990 100644
--- a/libc/string/arc/memcmp.S
+++ b/libc/string/arc/memcmp.S
@@ -1,6 +1,7 @@
 /* Copyright (C) 2007 ARC International (UK) LTD */
 
 #include <bits/asm.h>
+#include <features.h>
 
 #ifdef __LITTLE_ENDIAN__
 #define WORD2 r2
@@ -115,3 +116,7 @@ ENTRY(memcmp)
 	j_s.d	[blink]
 	mov	r0,0
 ENDFUNC(memcmp)
+
+#ifdef __UCLIBC_SUSV3_LEGACY__
+strong_alias(memcmp,bcmp)
+#endif
-- 
1.8.1.2

